src/main/java/io/github/genomicdatainfrastructure/discovery/services/BeaconFilteringTermsService.java

```diff
@@ -0,0 +1,122 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package io.github.genomicdatainfrastructure.discovery.services;
+
+import static java.util.Optional.ofNullable;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.mapping;
+import static java.util.stream.Collectors.toMap;
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import io.github.genomicdatainfrastructure.discovery.model.Facet;
+import io.github.genomicdatainfrastructure.discovery.model.FacetGroup;
+import io.github.genomicdatainfrastructure.discovery.model.ValueLabel;
+import io.github.genomicdatainfrastructure.discovery.remote.beacon.api.BeaconQueryApi;
+import io.github.genomicdatainfrastructure.discovery.remote.beacon.model.BeaconFilteringTerm;
+import io.github.genomicdatainfrastructure.discovery.remote.beacon.model.BeaconFilteringTermsResponse;
+import io.github.genomicdatainfrastructure.discovery.remote.beacon.model.BeaconFilteringTermsResponseContent;
+import io.github.genomicdatainfrastructure.discovery.remote.beacon.model.BeaconResource;
+import io.quarkus.cache.CacheResult;
+import jakarta.enterprise.context.ApplicationScoped;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+@ApplicationScoped
+public class BeaconFilteringTermsService {
+
+    public static final String BEACON_FACET_GROUP = "beacon";
+
+    private static final String DEFAULT_SCOPE = "individual";
+
+    private final BeaconQueryApi beaconQueryApi;
+
+    public BeaconFilteringTermsService(
+            @RestClient BeaconQueryApi beaconQueryApi
+    ) {
+        this.beaconQueryApi = beaconQueryApi;
+    }
+
+    @CacheResult(cacheName = "beacon-facet-group-cache")
+    public FacetGroup listFilteringTerms(String authorization) {
```

Comment:
Ah okay, so this endpoint can be used to list filtering terms to frontend?

Reply:
Yep, the filtering terms/filters/facets will be present in the `/api/v1/datasets/search` response. The only downside of this new endpoint is the response payload. It's quite big and the Beacon API doesn't provide ways to reduce or filter the response content. That's why we will cache the results.